superset/common/query_context_processor.py

```diff
@@ -267,17 +271,28 @@ def ensure_totals_available(self) -> None:
             is_totals_query = (
                 not query.columns and query.metrics and not query.post_processing
             )
-            if is_totals_query:
-                totals_queries.append(i)
+            if is_totals_query and totals_idx is None:
+                totals_idx = i
+
+        if queries_needing_totals and totals_idx is not None:
+            totals_query = self._query_context.queries[totals_idx]
+            totals_query.row_limit = None
```

Comment:
I understand this is not the scope of the PR (previously we were already doing this) but do you know why the query gets here with a `row_limit`? I would imagine the client/chart to send the query correctly?

Reply:
I don't think it makes a difference in practice, but this is because if we compute the totals for a query like this:

```sql
  SELECT region, SUM(sales) as sales
  FROM orders
  GROUP BY region
  LIMIT 50000
```

We want the totals query to not have a limit:

```sql
  SELECT SUM(sales) as total_sales
  FROM orders
```

Because the cardinality of the second query is always `<=` the cardinality of the first the limit shouldn't matter.